std/typecons.d

```diff
@@ -997,6 +1002,10 @@ template Tuple(Specs...)
             auto s = a.slice!(1, 3);
             static assert(is(typeof(s) == Tuple!(string, float)));
             assert(s[0] == "abc" && s[1] == 4.5);
+
+            // Phobos issue #15645
+            Tuple!(int, short, bool, double) b;
+            static assert(!__traits(compiles, b.slice!(2, 4)));
```

Comment:
I don't know much about this code, but isn't the `4` out of bounds? Or are the arguments 1-indexed?